docs/configuration.md

```diff
@@ -176,6 +176,7 @@ The following settings control pyright’s diagnostic output (warnings or errors
 
 **reportMatchNotExhaustive** [boolean or string, optional]: Generate or suppress diagnostics for a 'match' statement that does not provide cases that exhaustively match against all potential types of the target expression. The default value for this setting is 'none'.
 
+**reportShadowedImports** [boolean or string, optional]: Generate or suppress diagnostics for files that are overriding a module in the stdlib. The default value for this setting is 'none'.
```

Comment:
Will do

Reply:
It's actually not off for basic/strict. Do you think it should be?

Reply:
Hmm, I'm not sure this should be enabled by default anywhere. If it is enabled, I would think that it should be a warning not an error, and it should be overridable in the config (unlike most strict-mode rules that cannot be overridden — see `getStrictModeNotOverriddenRules`).

I'm not fully sold on this check to begin with. Does it belong in a type checker? It's not related to type checking, after all. It's more of a linting feature. Having internal modules with the same name as stdlib modules is supported by Python. It may be undesirable and lead to problems if you don't know what you're doing, but it is legit.

Reply:
I agree error sounds like too much. 

I think the main reason it's a request on pylance is because it was something the jupyter extension implemented on its own but didn't really have enough information to do a good job with. The goto def part is hard without writing an import resolver and all of the rest of the infrastructure in pylance. 

I'll turn it off in pyright.

Reply:
And I don't think we wanted to wait/add this to a different linter since it was such a common problem in the jupyter extension.

Reply:
How do I know where pyright is finding the stdlib?

Reply:
Docs on where pyright looks for python paths are here: https://github.com/microsoft/pyright/blob/main/docs/configuration.md

Pylance uses the current interpreter selected in VS code to figure out the stdlib path.